lib/saml/settings_service.rb

```diff
@@ -0,0 +1,20 @@
+module SAML
+  # This class is responsible for putting together a complete ruby-saml
+  # SETTINGS object, meaning, our static SP settings + the IDP settings
+  # which must be fetched once and only once via IDP metadata.
+  require 'singleton'
+  class SettingsService
+    include Singleton
+
+    def saml_settings
+      @combined_saml_settings ||= fetch_idp_metadata
+    end
+
+    ## Makes an external web call to get IDP metadata and populates SETTINGS
+    def fetch_idp_metadata
+      parser = OneLogin::RubySaml::IdpMetadataParser.new
+      parser.parse_remote(SAML_CONFIG['metadata_url'], true, settings: SETTINGS)
+      SETTINGS
```

Comment:
I'm not sure I agree with that. Seeing as SAML is a service, the first place I would look is in the lib/saml folder where all of our other services are. We have configurations for those services in configuration classes, see lib/sm and lib/rx for example. They only reference the HOSTNAME from application.yml